src/libraries/System.Private.CoreLib/src/System/IO/RandomAccess.Unix.cs

```diff
@@ -0,0 +1,149 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.Buffers;
+using System.Collections.Generic;
+using System.IO.Strategies;
+using System.Runtime.InteropServices;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.Win32.SafeHandles;
+
+namespace System.IO
+{
+    public static partial class RandomAccess
+    {
+        // IovStackThreshold matches Linux's UIO_FASTIOV, which is the number of 'struct iovec'
+        // that get stackalloced in the Linux kernel.
+        private const int IovStackThreshold = 8;
+
+        internal static long GetFileLength(SafeFileHandle handle, string? path)
+        {
+            int result = Interop.Sys.FStat(handle, out Interop.Sys.FileStatus status);
+            FileStreamHelpers.CheckFileCall(result, path);
+            return status.Size;
+        }
+
+        private static unsafe int ReadAtOffset(SafeFileHandle handle, Span<byte> buffer, long fileOffset)
```

Comment:
I'd prefer if we not use the scatter/gather and atoffset terminology in private/internal members either.

Reply:
I see your point, but we need two method names per single operation:

-  public method that validates the input in OS-agnostic way
-  an OS-specific implementation that accepts the same arguments as the public method

I believe that using `offset|scatter|gather` terminology is better than using prefixes like `Core` or `Internal`:

```cs
public int Read(SafeFileHandle handle, Span<byte> buffer, long fileOffset)
{
    // validate the input

   return ReadInternal(handle, buffer, offset);
}

internal int ReadInternal(SafeFileHandle handle, Span<byte> buffer, long fileOffset)
{
  // do the actual read
}
```

Reply:
> I believe that using offset|scatter|gather terminology is better than using prefixes like Core or Internal

FWIW, I disagree.  "Core" or "Internal" is used to indicate when something is the meaningful implementation of something else.  Using "Offset/Scatter/Gather" suggests to me it's an entirely different implementation from the public shim wrapping it.